.github/workflows/deploy_to_environment.yml

```diff
@@ -42,25 +45,40 @@ jobs:
           github-token: ${{ secrets.GITHUB_TOKEN }}
           snyk-token: ${{ secrets.SNYK_TOKEN }}
 
+
   deploy:
     name: Deploy to environment
-    needs: [docker]
+    concurrency: deploy_${{ inputs.environment }}
+    needs: docker
+    if: |
+      always() && 
+      (needs.docker.result == 'skipped' || needs.docker.result == 'success' )
     runs-on: ubuntu-latest
-    environment: ${{ inputs.environment }}
-    outputs:
-      docker-image: ${{ needs.docker.outputs.docker-image }}
+    environment:
+      name: ${{ inputs.environment }}
     steps:
       - uses: actions/checkout@v4
         with:
-          ref: ${{ inputs.branch-name }}
+          ref: ${{ github.event.inputs.sha }}
+
+      - name: setup environment variables
+        id: setup-env
+        run: |
+          if [[ "${{ github.event.inputs.build }}" != "false" ]]; then
+            echo "IMAGE_TAG=${{ needs.docker.outputs.docker-image }}" >> $GITHUB_ENV
+            echo "COMMIT_SHA=${{ github.sha }}" >> $GITHUB_ENV
```

Comment:
echo "COMMIT_SHA=${{ needs.docker.outputs.image-tag }}" >> $GITHUB_ENV
would seem clearer to me